layer4/matchers.go

```diff
@@ -28,6 +29,7 @@ func init() {
 
 // ConnMatcher is a type that can match a connection.
 type ConnMatcher interface {
+	caddy.Module
```

Comment:
To make sure that `CaddyModule()` can be called in the [Match](https://github.com/mholt/caddy-l4/pull/71/files#diff-df4c4d5d1b7d95094092aea25682cc4da237cc69096bf62f2497a9285c070e86R55) function to extract the matcher name.

Reply:
Ah, now I see it.

I wonder, if that could be optional. It doesn't _strictly_ have to be a Caddy module (though almost always will be!) and since it's just for logging I wonder if a type assertion could be used to optionally get that information. That might be better than requiring another interface here.

Reply:
Yes I can change this. But what should the logged matcher name be if the type assertion fails? Just `unknown` ? Without the `matcher` field the log message barely makes sense but omitting it would be even worse for debugging.

Reply:
@ydylla You can use the `.With()` method to conditionally add the field, or even better yet, I think the zap logger omits serializing empty field values. (I know it omits nil, not sure about empty.)

Reply:
Yes but I think reading just:
`debug layer4 matching {"remote": "10.0.0.2:35720", "matched": false}`
is not very helpful. I would prefer:
`debug layer4 matching {"remote": "10.0.0.2:35720", "matcher": "unknown", "matched": false}`
which makes it clear the name could not be extracted and one maybe want to look into that as well.